interfaces/datastructure/Frame.h

```diff
@@ -58,10 +58,18 @@ class SOLARFRAMEWORK_API Frame {
 	/// @return view image
     const SRef<Image>& getView() const;
 
+    /// @brief get mask
+    /// @return mask
+    const SRef<Image>& getMask() const;
```

Comment:
* Why the `&`? -> return SRef by value to garantee proper counter increment.

```cpp
const SRef<Image> getMask() const;
```

* Also, as it is often the case, here `const` does not apply to Image, but to the SRef. Is that what's really intended? If the constness concerns the Image, then use `SRef<const Image>` (or even `const SRef<const Image>`, but it seems overkill)

* Returning a shared pointer means that we allow the mask Image to outlive the Frame. Is that necessary? If not, just return a (const) ref?
```cpp
[const] Image& getMask() const;
```

It all depends on how you intend it to be used elsewhere (I haven't yet seen that).